Sources/Widgets/Manipulators/CPRManipulator/index.js

```diff
@@ -0,0 +1,140 @@
+
+  publicAPI.handleEvent = (callData, glRenderWindow) => {
+    const mapper = model.cprActor?.getMapper();
+    if (!mapper) {
+      return { worldCoords: [0, 0, 0] };
+    }
+
+    // Get normal and origin of the picking plane from the actor matrix
+    const cprActorMatrix = [];
+    mat4.transpose(cprActorMatrix, model.cprActor.getMatrix());
+
+    const worldPlaneNormal = cprActorMatrix.slice(8, 11); // 3rd column
+    const worldPlaneOrigin = cprActorMatrix.slice(12, 15); // 4th column
+
+    // Convert world plane position to 2D position in the plane
+    const inversecprActorMatrix = [];
+    mat4.invert(inversecprActorMatrix, cprActorMatrix);
+    const worldPlanePicking = intersectDisplayWithPlane(
+      callData.position.x,
+      callData.position.y,
+      worldPlaneOrigin,
+      worldPlaneNormal,
+      callData.pokedRenderer,
+      glRenderWindow
+    );
+    const modelPlanePicking = []; // (x, height - distance, 0)
+    vec3.transformMat4(
+      modelPlanePicking,
+      worldPlanePicking,
+      inversecprActorMatrix
+    );
+    const height = mapper.getHeight();
+    const distance = height - modelPlanePicking[1];
+
+    return publicAPI.distanceEvent(distance);
+  };
+
+  publicAPI.distanceEvent = (distance) => {
+    const mapper = model.cprActor?.getMapper();
+    if (!mapper) {
+      return { worldCoords: [0, 0, 0] };
+    }
+    const height = mapper.getHeight();
+    const clampedDistance = Math.max(0, Math.min(height, distance));
+    const { position, orientation } =
+      mapper.getCenterlinePositionAndOrientation(clampedDistance);
+
+    let worldDirection;
+    if (orientation) {
+      const modelDirections = mat3.fromQuat([], orientation);
+      const baseDirections = mapper.getDirectionMatrix();
+      worldDirection = mat3.mul([], modelDirections, baseDirections);
+    }
+
+    model.currentDistance = clampedDistance;
+    return { worldCoords: position, worldDirection };
+  };
+
+  publicAPI.handleScroll = (nbSteps) => {
+    const distance = model.currentDistance + model.distanceStep * nbSteps;
+    return publicAPI.distanceEvent(distance);
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+// currentDistance is the distance from the first point of the centerline
+// cprActor.getMapper() should be a vtkImageCPRMapper
+function defaultValues(initialValues) {
+  const values = {
+    distanceStep: 1,
+    currentDistance: 0,
+    cprActor: null,
+    ...initialValues,
+  };
+  // Find default distanceStep from image spacing
+  // This only works if the mapper in the actor already has an ImageData
+  if (!initialValues.distanceStep) {
+    const imageSpacing = initialValues.cprActor
+      ?.getMapper()
+      ?.getInputData(0)
+      ?.getSpacing?.();
+    if (imageSpacing) {
+      values.distanceStep = Math.min(...imageSpacing);
```

Comment:
this should not be evaluated at initialization time, but each time distanceStep is required. Because a `setDistanceStep()` could set `model.distanceStep` to 0 or null at anytime.